### PR TITLE
Nomenclature: Prefer 'your code' over 'your app'

### PR DESCRIPTION
**Pull request by perlun**
I happened to notice this while reading the docs at https://docs.microsoft.com/en-us/dotnet/api/System.Text.StringBuilder?view=net-5.0&viewFallbackFrom=netcore-5.0. The text consistently talked about your "app" in multiple places.

While this might seem like a nice and modern way to express things, I do think that it conveys the wrong meaning to the reader. This is because "app" has a specific meaning - often used as synonym to _mobile application_. While a webapp could also reasonably be called an "app", probably not so much for your random console application, your Windows background service, your Linux daemon... etc. These are not "apps", but `StringBuilder` is still perfectly usable in such scenarios also.

In other words, it feels more correct in this context to talk about "application" since it has a broader scope. In other words, all _apps_ are applications, but all _applications_ are not apps... :smile:

### CODE REVIEW

**Comment by tdykstra** [comment]
cc @Rick-Anderson

**Comment by gewarren** [comment]
The MS style guide says:

In general, both app and application are acceptable. App is preferred in names and when space is limited.

**Comment by perlun** [comment]
Thanks for your reply @gewarren. Do you happen to have a URL to the referred-to style guide?

> In general, both app and application are acceptable. App is preferred in names and when space is limited.

As for the latter part, neither point is true in this particular case. I think clarity is more important than conciseness in this context.

**Comment by gewarren** [comment]
> Thanks for your reply @gewarren. Do you happen to have a URL to the referred-to style guide?
> 
> > In general, both app and application are acceptable. App is preferred in names and when space is limited.
> 
> As for the latter part, neither point is true in this particular case. I think clarity is more important than conciseness in this context.

Here's the link, but I think it's internal to MS: https://styleguides.azurewebsites.net/Styleguide/Read?id=2696&topicid=27896.

I wasn't necessarily opining for one or the other, just adding info from the style guide.

**Comment by adegeo** [comment]
I prefer `app`. When trying to help a friend or family member troubleshoot something on their computer, I don't use the word "application" I use "app" and they know what I'm talking about. It might have had a decade of being a synonym for "mobile application" but if you look at the word app it's been used in desktop computer for much longer.

The word "application" doesn't even appear on https://www.microsoft.com/en-us/windows/windows-10-apps

Application sounds formal these days, and probably has its place. It's the evolution of common computer terms. We don't use E-Mail anymore, it's email.

**Comment by perlun** [comment]
@adegeo

> I prefer `app`. When trying to help a friend or family member troubleshoot something on their computer, I don't use the word "application" I use "app" and they know what I'm talking about.

Sure, I get that point, but I'm not sure how relevant this is in this context though. This is dealing with how the `StringBuilder` .NET class is documented and the target audience for these docs can be assumed to be slightly different than in the scenario you describe.

However, it _is_ indeed a challenge between at least the following goals:

- avoid "dumbing things down" too much, and
- avoid "speaking above the heads" of people

Both of these are obviously important goals but I would perhaps (for this context) consider the former being the more important design consideration here. Feel free to disagree of course.

> It might have had a decade of being a synonym for "mobile application" but if you look at the word app, it's been used in desktop computing prior to that.

Thanks for pointing this out and digging up a classic example from the good old days. :+1: :heart: I would _guess_ (don't have any facts on this though) that this usage was less common than "application" back then, though. I think it was largely popularized by another company whose name coincidentally also starts with "app" around the time of 2007 and onwards. :wink: 

This Stack Exchange thread was quite interesting, even though it's written a few years ago: https://english.stackexchange.com/questions/45762/appropriate-use-of-app-vs-application. Here are some quotes from that thread:

> Gartner says: “Apps and applications are two very different expressions of software. The defining characteristic of an app is its reduced functional presence. Apps do less than applications. That is their goal."

> Apple and Google talk about "Apps" even in their formal docs, because they've made a conscious choice to adopt that into their company lexicon. Is that appropriate for all companies? I think the short answer is: it depends. Consider the domain (mobile apps versus windows apps) **and the tech-savvy sophistication of the audience.** [my emphasis]

The last sentence reiterates my point above - it depends on what kind of readers of this text we are expecting and their level of technical experience.

**Comment by Rick-Anderson** [comment]
> The MS style guide says:
> 
> In general, both app and application are acceptable. App is preferred in names and when space is limited.

app, application
The Microsoft technologies that customers use to get things done come in a variety of shapes and sizes. Always refer to your group's editorial style guide for details about when to use app, application, product, and service.

If possible, refer to a product by its descriptor, such as database management system or spreadsheet.

***Most of the time use app, not application***, to describe desktop apps and cloud apps for tablets and mobile devices.

Global tip The abbreviated form, app, isn't available in all languages, so allow enough space for the full translated term in localized content.

For full-featured products, such as Office and Visual Studio, it's OK to use program instead of application.

It's OK to use application in content for a technical audience, especially to refer to a set of software files that includes executable files and other components.

Don't use application program.

https://styleguides.azurewebsites.net/StyleGuide/Read?id=2700&topicid=32507

**Comment by perlun** [comment]
Thanks for the style guide excerpt @Rick-Anderson. :+1: 

> _**Most of the time use app, not application**_, to describe desktop apps and cloud apps for tablets and mobile devices.

Exactly, this sums it all up in a single sentence. _"To describe desktop apps and cloud apps **for tablets and mobile devices**"._ The .NET platform in (general and `StringBuilder` in particular) is not limited only to these use cases.

I'm personally using it to write console applications targeting Linux and other command-line oriented scenarios. There's no GUI involved. It feels overly limiting to talk about "app" in that context, since it slightly implies (at least to me) a more narrow scope for this framework class than actually is the case here.

For more GUI-oriented classes (e.g. WPF, Windows Forms etc) I think it's quite fine to talk about "app". For things more generic in nature, used for both GUI and non-GUI applications alike, I still insist that "application" is better for getting the message across.

**Comment by adegeo** [comment]
Just more info. I noticed today that the templates in Visual Studio for a new desktop app say

- `Windows Forms App`
- `WPF Application`

No one can decide 🤣

**Comment by Rick-Anderson** [comment]
> Apple and Google talk about "Apps" even in their formal docs, because they've made a conscious choice to adopt that into their company lexicon. 

We've never had a complaint about using app for web applications.

> This Stack Exchange thread was quite interesting

SE is very opinionated. I disagree with that assertion. Goog and APL got it right.

**Comment by adegeo** [comment]
_My final opinion on the matter..._

@perlun Scanning the article, there were many instances where it talked about "the app/application" but really it's talking about your code and doesn't need to call out an app. I would say if you wanted to remove app/application where it made sense and change it to a more generic term of code or program, go for it. But as @Rick-Anderson was pointing out, we probably don't need to choose one term over the other regarding app/application; people aren't going to be confused with app vs application.

**Comment by perlun** [comment]
@adegeo 

> @perlun Scanning the article, there were many instances where it talked about "the app/application" but really it's talking about your code and doesn't need to call out an app. I would say if you wanted to remove app/application where it made sense and change it to a more generic term of code or program, go for it. But as @Rick-Anderson was pointing out, we probably don't need to choose one term over the other regarding app/application; people aren't going to be confused with app vs application.

Thanks, I think this might be some valid points you're making here. :+1: I edited the PR and amended it to speak about `your program`, to avoid the whole app vs application debate altogether. Please give it a look and see if you think it is an improvement from the original wording; if so, we could consider moving forward with it.

File: xml/System.Text/StringBuilder.xml [hunk 1]
```diff
@@ -105,21 +105,21 @@
...
->  Although the <xref:System.Text.StringBuilder> class generally offers better performance than the <xref:System.String> class, you should not automatically replace <xref:System.String> with <xref:System.Text.StringBuilder> whenever you want to manipulate strings. Performance depends on the size of the string, the amount of memory to be allocated for the new string, the system on which your app is executing, and the type of operation. You should be prepared to test your app to determine whether <xref:System.Text.StringBuilder> actually offers a significant performance improvement.
+>  Although the <xref:System.Text.StringBuilder> class generally offers better performance than the <xref:System.String> class, you should not automatically replace <xref:System.String> with <xref:System.Text.StringBuilder> whenever you want to manipulate strings. Performance depends on the size of the string, the amount of memory to be allocated for the new string, the system on which your program is executing, and the type of operation. You should be prepared to test your program to determine whether <xref:System.Text.StringBuilder> actually offers a significant performance improvement.
```
Comment (Rick-Anderson) [inline xml/System.Text/StringBuilder.xml:108]: We consistently use app, sometimes application. We rarely use program. App is the standard we use.

Comment (adegeo) [reply xml/System.Text/StringBuilder.xml:108]: A lot of these would use code. This one specifically though I would say should remain app.

File: xml/System.Text/StringBuilder.xml [hunk 2]
```diff
@@ -105,21 +105,21 @@
...
 
  Consider using the <xref:System.String> class under these conditions:
 
--   When the number of changes that your app will make to a string is small. In these cases, <xref:System.Text.StringBuilder> might offer negligible or no performance improvement over <xref:System.String>.
+-   When the number of changes that your program will make to a string is small. In these cases, <xref:System.Text.StringBuilder> might offer negligible or no performance improvement over <xref:System.String>.
```
Comment (adegeo) [inline xml/System.Text/StringBuilder.xml:112]: This type of wording I would use `code` instead. it could be a library or an app really. But in general, most of this all talks about the code you're writing.

**Comment by perlun** [comment]
@adegeo and others, sorry for the delay but the PR is now updated & rebased on top of the latest `main`. Maybe this would be a slight improvement to the current wording.

File: xml/System.Text/StringBuilder.xml [hunk 3]
```diff
@@ -783,7 +783,7 @@
  The <xref:System.Text.StringBuilder.Append%28System.Boolean%29> method modifies the existing instance of this class; it does not return a new class instance. Because of this, you can call a method or property on the existing reference and you do not have to assign the return value to a <xref:System.Text.StringBuilder> object, as the following example illustrates.
 
  :::code language="csharp" source="~/samples/snippets/csharp/VS_Snippets_CLR_System/system.Text.StringBuilder.Append/cs/append2.cs" interactive="try-dotnet-method" id="Snippet2":::
- :::code language="vb" source="~/samples/snippets/visualbasic/VS_Snippets_CLR_System/system.Text.StringBuilder.Append/vb/append2.vb" id="Snippet2"::: 
+ :::code language="vb" source="~/samples/snippets/visualbasic/VS_Snippets_CLR_System/system.Text.StringBuilder.Append/vb/append2.vb" id="Snippet2":::
```
Comment (perlun) [inline xml/System.Text/StringBuilder.xml:786]: (A few whitespace fixes like this slipped through, there were some trailing blanks and my VS Code is configured to trim blanks on save. Can revert these/move to a separate PR if you so prefer.)

**Review by tdykstra — APPROVED** [review]
Thanks, @perlun !
